### PR TITLE
MCR-2707 Improve ObjectIDListing

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2707).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntity.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    private Instant modifyDate;
+
+    private Instant deleteddate;
+
+    @Id
+    @Column(name = "objectid", length = MCRObjectID.MAX_LENGTH, nullable = false)
+    @Basic
+    public String getObjectId() {
+        return objectId;
+    }
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntity.java]: Should be MCRObjectID not String.

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntity.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    @Column(name = "objectnumber")
+    public int getObjectNumber() {
+        return objectNumber;
+    }
+
+    public void setObjectNumber(int objectNumber) {
+        this.objectNumber = objectNumber;
+    }
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntity.java]: the getter should use the methods of MCRObjectID

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Comment by yagee-de** [comment]
Still not done with review but a first suggestion. Sorry

**Comment by yagee-de** [comment]
We need more JUnit test and javadocs

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntity.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+import org.mycore.datamodel.metadata.MCRObjectID;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
+
+@Entity
+@IdClass(MCRObjectIDPK.class)
+public class MCRObjectEntity {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntity.java]: Naming? MCRObjectInfo was a suggestion

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityCommands.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import org.mycore.datamodel.common.MCRXMLMetadataManager;
+import org.mycore.datamodel.metadata.MCRMetadataManager;
+import org.mycore.datamodel.metadata.MCRObject;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.frontend.cli.annotation.MCRCommand;
+import org.mycore.frontend.cli.annotation.MCRCommandGroup;
+import org.xml.sax.SAXException;
+
+@MCRCommandGroup(name = "Object Entity Commands")
+public class MCRObjectEntityCommands {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityCommands.java]: rename accordingly

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityEventHandler.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+package org.mycore.backend.jpa;
+
+import java.time.Instant;
+
+import org.mycore.common.MCRSessionMgr;
+import org.mycore.common.events.MCREvent;
+import org.mycore.common.events.MCREventHandlerBase;
+import org.mycore.datamodel.metadata.MCRObject;
+
+public class MCRObjectEntityEventHandler extends MCREventHandlerBase {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityEventHandler.java]: rename accordingly

Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityEventHandler.java]: Add Javadoc

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityManager.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.Optional;
+
+import org.mycore.datamodel.classifications2.MCRCategoryID;
+import org.mycore.datamodel.metadata.MCRObject;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.datamodel.metadata.MCRObjectService;
+
+import jakarta.persistence.EntityManager;
+
+public class MCRObjectEntityManager {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityManager.java]: rename accordingly
add Javadoc

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityManager.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            return;
+        }
+        MCREntityManagerProvider.getCurrentEntityManager().remove(byID);
+    }
+
+    /**
+     * Updates the state of the entity to be deleted (does not {@link #remove(MCRObject)} the object)
+     * @param object
+     */
+    public static void delete(MCRObject object, Instant deletedDate, String deletedBy) {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityManager.java]: Only public method is to delete?

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityQueryResolver.java [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import jakarta.persistence.metamodel.SingularAttribute;
+
+public class MCRObjectEntityQueryResolver implements MCRObjectQueryResolver {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectEntityQueryResolver.java]: rename accordingly and add Javadocs and JUnit tests

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectIDPK.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import org.mycore.datamodel.metadata.MCRObjectID;
+
+import jakarta.persistence.Access;
+import jakarta.persistence.AccessType;
+import jakarta.persistence.Basic;
+import jakarta.persistence.Column;
+import jakarta.persistence.Convert;
+
+@Access(AccessType.FIELD)
+public class MCRObjectIDPK implements Serializable {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectIDPK.java:63]: Javadocs ;-)

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRObjectQuery.java [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.mycore.datamodel.metadata.MCRObjectID;
+
+/**
+ * Used to Query a collection of mycore objects.
+ */
+public class MCRObjectQuery {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRObjectQuery.java:30]: Javadocs \@see ThisOtherClass

File: mycore-base/src/main/resources/config/mycore.properties [hunk 11]
```diff
@@ -114,6 +114,7 @@ MCR.Access.AddObjectDefaultRule=true
 # MCR.EventHandler.MCRObject.017.Class=org.mycore.datamodel.common.MCRServiceFlagEventHandler
 # Define XML store handler
   MCR.EventHandler.MCRObject.020.Class=org.mycore.datamodel.common.MCRXMLMetadataEventHandler
+  MCR.EventHandler.MCRObject.020A.Class=org.mycore.backend.jpa.MCRObjectEntityEventHandler
```
Comment (yagee-de) [inline mycore-base/src/main/resources/config/mycore.properties]: Anything better than officially use 020A? Do we have space left?

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectIDPK.java (hunk 9 above)
Comment (sebhofmann) [reply mycore-base/src/main/java/org/mycore/backend/jpa/MCRObjectIDPK.java:63]: You wrote that Class: https://github.com/MyCoRe-Org/mycore/pull/1632/commits/b69cc1f992de52bdda4a798f3c93493511db76ee

**Pull request closed without merging**